src/test/java/com/google/jenkins/plugins/computeengine/integration/ITUtil.java

```diff
@@ -0,0 +1,200 @@
+import java.util.List;
+import java.util.Map;
+import java.util.logging.LogManager;
+import java.util.logging.Logger;
+import java.util.logging.StreamHandler;
+import org.jvnet.hudson.test.JenkinsRule;
+
+class ITUtil {
+
+  static final String DEB_JAVA_STARTUP_SCRIPT = "#!/bin/bash\n" +
+      "/etc/init.d/ssh stop\n" +
+      "echo \"deb http://http.debian.net/debian stretch-backports main\" | \\\n" +
+      "      sudo tee --append /etc/apt/sources.list > /dev/null\n" +
+      "apt-get -y update\n" +
+      "apt-get -y install -t stretch-backports openjdk-8-jdk\n" +
+      "update-java-alternatives -s java-1.8.0-openjdk-amd64\n" +
+      "/etc/init.d/ssh start";
+
+  private static final String CLOUD_NAME = "integration";
+  private static final String NAME_PREFIX = "integration";
+  private static final String REGION = format("projects/%s/regions/us-west1");
+  static final String ZONE = "us-west1-a";
+  private static final String ZONE_BASE = format("projects/%s/zones/" + ZONE);
+  static final String LABEL = "integration";
+  private static final String MACHINE_TYPE = ZONE_BASE + "/machineTypes/n1-standard-1";
+  static final String NUM_EXECUTORS = "1";
+  private static final boolean PREEMPTIBLE = false;
+  //  TODO: Write a test to see if min cpu platform worked by picking a higher version?
+  private static final String MIN_CPU_PLATFORM = "Intel Broadwell";
+  private static final String CONFIG_DESC = "integration";
+  private static final String BOOT_DISK_TYPE = ZONE_BASE + "/diskTypes/pd-ssd";
+  private static final boolean BOOT_DISK_AUTODELETE = true;
+  private static final String BOOT_DISK_PROJECT_ID = "debian-cloud";
+  private static final String BOOT_DISK_IMAGE_NAME = "projects/debian-cloud/global/images/family/debian-9";
+  private static final String BOOT_DISK_SIZE_GB_STR = "10";
+  private static final Node.Mode NODE_MODE = Node.Mode.EXCLUSIVE;
+  private static final String ACCELERATOR_NAME = "";
+  private static final String ACCELERATOR_COUNT = "";
+  private static final String RUN_AS_USER = "jenkins";
+  static final String NULL_TEMPLATE = null;
+  private static final String NETWORK_NAME = format("projects/%s/global/networks/default");
+  private static final String SUBNETWORK_NAME = "default";
+  private static final boolean EXTERNAL_ADDR = true;
+  private static final String NETWORK_TAGS = "ssh";
+  private static final String SERVICE_ACCOUNT_EMAIL = "";
+  private static final String RETENTION_TIME_MINUTES_STR = "";
+  private static final String LAUNCH_TIMEOUT_SECONDS_STR = "";
+
+  static String projectId = format("%s");
+
+  private static String format(String s) {
+    String projectId = System.getenv("GOOGLE_PROJECT_ID");
+    if (projectId == null) {
+      throw new RuntimeException("GOOGLE_PROJECT_ID env var must be set");
+    }
+    return String.format(s, projectId);
+  }
+
+  static ComputeClient init(JenkinsRule r, StreamHandler sh, Map<String, String> label, Logger log) throws Exception {
```

Comment:
This method groups together a lot of separate concerns, suggest breaking each unit of functionality into an orthogonal method. Additional, there are a lot of side-effects occurring which result in tight-coupling, suggest taking a more functional approach by returning things like the Cloud which is initialized, and allow callers to add hooks into the JenkinsRule, (the logic for which could verify well exist in an orthogonal method for that purpose).